### ISSUE TITLE
Ruby Gem: Load sections of a page after initial load by wrapping them in a single block

### ISSUE THREAD

**Issue by johnsonj** [thechangelog/ping#536]
Hello!

Suggesting my project, [progressive_render](https://github.com/johnsonj/progressive_render), for the changelog. The project aims to be a very low-touch solution to progressive rendering web pages. Developers wrap slow or less important content in a block and the gem handles the rest.

![Screencast Demo](https://camo.githubusercontent.com/a83beb2284f0fd02946ca247d708bb9c40f1e755/687474703a2f2f692e696d6775722e636f6d2f4a3952746244632e676966)

# Basic Usage
```ruby
<%=progressive_render do %>
    <h1>Content!</h1>
    <% sleep 5 %>
<% end %>
```

Thanks,
Jeff

**Comment by jerodsanto** [comment]
@johnsonj cool stuff! How does it work? I mean, I understand it makes ajax calls for the less-important content, but does the controller then render those as separate partials or something?

**Comment by johnsonj** [comment]
The controller renders the entire page (minus other `progressive_render` blocks) when the ajax request goes to fetch the missing content and extracts out only the interested block.

**Comment by jerodsanto** [comment]
I see, makes sense. So it skips rendering them the first time through, then on subsequent requests it re-renders the entire thing PLUS the specific `progressive_render` block that is fetching itself. Neat.

We'll include this in this weekend's issue of [Changelog Weekly](https://changelog.com/weekly) 🍻

**Comment by johnsonj** [comment]
Awesome, thanks for checking it out Jerod!

On Fri, Oct 14, 2016, 10:54 AM Jerod Santo <notifications@github.com> wrote:

Closed #536 <https://github.com/thechangelog/ping/issues/536>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/thechangelog/ping/issues/536#event-824336512>, or mute
the thread
